lib/nodejs/esm-utils.js

```diff
@@ -39,48 +39,34 @@ exports.requireOrImport = async (file, esmDecorator) => {
     return formattedImport(file, esmDecorator);
   }
   try {
-    return dealWithExports(await formattedImport(file, esmDecorator));
-  } catch (err) {
+    return require(file);
+  } catch (requireErr) {
     if (
-      err.code === 'ERR_MODULE_NOT_FOUND' ||
-      err.code === 'ERR_UNKNOWN_FILE_EXTENSION' ||
-      err.code === 'ERR_UNSUPPORTED_DIR_IMPORT'
+      requireErr.code === 'ERR_REQUIRE_ESM' ||  requireErr.code === 'ERR_INTERNAL_ASSERTION' ||
+      (requireErr instanceof SyntaxError &&
+        requireErr
+          .toString()
+          .includes('Cannot use import statement outside a module'))
     ) {
+      // In Node.js environments after version 22.11, the `loadESMFromCJS` function 
+      // is used within the `require` function to handle cases where the target file 
+      // is in ESM (ECMAScript Module) format. If the Node.js version is after 22.11, 
+      // the code will import the module without any issues. For older versions, 
+      // this `if` statement is required to properly handle ESM modules.
```

Comment:
[Docs] Phrases like _"required to properly handle ..."_ are often indications the docs are unclear. The fact that the code's author believed it required to handle something is implied by the code existing. What readers really need to know is _why_ it's required.

```suggestion
      // the code will import the module without any issues.
      // For older Node.js versions, we fall back to trying an ESM `import()`.
      // Note that the `import()` must come after the `require()` to avoid a cyclic dependency.
      // See https://github.com/mochajs/mocha/issues/5290.
```
```